finesse/hardware/st10_controller.py

```diff
@@ -0,0 +1,343 @@
+        """Create a new ST10Controller with the specified serial device properties.
+
+        Args:
+            port: Serial port name
+            baudrate: Serial port baudrate
+            timeout: How long to wait for read operations (seconds)
+            serial_args: Extra arguments to Serial constructor
+            serial_kwargs: Extra keyword arguments to Serial constructor
+        """
+        if "write_timeout" not in serial_kwargs:
+            serial_kwargs["write_timeout"] = timeout
+
+        serial = Serial(port, baudrate, *serial_args, timeout=timeout, **serial_kwargs)
+        return ST10Controller(serial)
+
+    def __del__(self) -> None:
+        """Leave mirror facing downwards when finished.
+
+        This prevents dust accumulating.
+        """
+        try:
+            self.stop_moving()
+            self.move_to("nadir")
+        except Exception as e:
+            logging.error(f"Failed to reset mirror to downward position: {e}")
+
+    def _check_device_id(self) -> None:
+        """Check that the ID is the correct one for an ST10.
+
+        Raises:
+            SerialException: Error communicating with device
+            SerialTimeoutException: Timed out waiting for response from device
+            ST10ControllerError: The device ID is not for an ST10
+        """
+        self._write("MV")
+        if self._read() != "107F024":
+            raise ST10ControllerError("Device ID indicates this is not an ST10")
+
+    def _get_input_status(self, index: int) -> bool:
+        """Read the status of the device's inputs.
+
+        The inputs to the controller include digital inputs, as well as other properties
+        like alarm status. The exact meaning seems to varies between boards.
+
+        Args:
+            index: Which boolean value in the input status array to check
+        """
+        input_status = self._request_value("IS")
+        return input_status[index] == "1"
+
+    @property
+    def steps_per_rotation(self) -> int:
+        """Get the number of steps that correspond to a full rotation."""
+        return self.STEPS_PER_ROTATION
+
+    def _home_and_reset(self) -> None:
+        """Return the stepper motor to its home position and reset the counter.
+
+        Raises:
+            SerialException: Error communicating with device
+            SerialTimeoutException: Timed out waiting for response from device
+            ST10ControllerError: Malformed message received from device
+        """
+        # If the third (boolean) value of the input status array is set, then move the
+        # motor first. I don't know what the input status actually means, but this is
+        # how it was done in the old program, so I'm copying it here.
+        if self._get_input_status(3):
+            self._relative_move(-5000)
+
+        # Send home command; leaves mirror facing upwards
+        self._write_check("SH6H")
+
+        # Turn mirror so it's facing down
+        self._relative_move(-30130)
```

Comment:
I still don't get why a relative rotation of 180 degrees is not STEPS_PER_ROTATION/2. I understand that this is what works, but still, it seems totally inconsistent. I'd suggest, when the system is available again, we run some tests about this and try to figure out why this is this way.

Reply:
It will be moving more than 180°, it's just that this is what the engineer wanted to define as the origin.

But yes, let's check it ourselves once we've got the kit.